gemrb/core/Scriptable/Actor.cpp

```diff
@@ -3463,38 +3463,46 @@ void Actor::UpdateFatigue()
 	if (!InParty || !game->GameTime) {
 		return;
 	}
-	// do icons here, so they persist for more than a tick
-	int LuckMod = core->ResolveStatBonus(this, "fatigue") ; // fatigmod.2da
-	if (LuckMod) {
-		AddPortraitIcon(PI_FATIGUE);
-	} else {
-		DisablePortraitIcon(PI_FATIGUE);
+
+	bool updated = false;
+	if (!TicksLastRested) {
+		// just loaded the game; approximate last rest
+		TicksLastRested = game->GameTime - (2*core->Time.hour_size) * (2*GetBase(IE_FATIGUE)+1);
+		updated = true;
+	} else if (LastFatigueCheck) {
+		ieDword FatigueDiff = (game->GameTime - TicksLastRested) / (4*core->Time.hour_size)
+		                    - (LastFatigueCheck - TicksLastRested) / (4*core->Time.hour_size);
```

Comment:
How can this ever be non-zero?

Reply:
Either right after a travel or exactly 4*n hours after a rest.  E.g. 4 hours after resting it evaluates to something like (4:00 / 4:00) - (3:59:59 / 4:00) = 1 - 0 (rounding) = 1.

Reply:
Ah, right. :+1: